src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,64 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+    private final Object[] keyArray;
+    private final Object[] valueArray;
+    private int arraySize = 0;
+
+    public StorageImpl() {
+        keyArray = new Object[MAX_ARRAY_LENGTH];
+        valueArray = new Object[MAX_ARRAY_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (arraySize >= MAX_ARRAY_LENGTH) {
+            throw new IllegalStateException("Storage is full. Maximum size is "
+                    + MAX_ARRAY_LENGTH + " .");
+        } else if (!updateValueByKey(key, value)) {
```

Comment:
```suggestion
        } 
        if (!updateValueByKey(key, value)) {
```
we don't need else after returning/throwing